spanner.go

```diff
@@ -109,37 +127,58 @@ func (h *spanner) getConstraints(q queryable) ([]spannerConstraint, error) {
 	}()
 
 	for rows.Next() {
-		var constraint spannerConstraint
+		var constraint SpannerConstraint
 		if err = rows.Scan(
-			&constraint.constraintName,
-			&constraint.referencingTable,
-			&constraint.foreignKeyColumn,
-			&constraint.referenceTable,
-			&constraint.referenceColumn,
+			&constraint.TableName,
+			&constraint.ConstraintName,
+			&constraint.ColumnName,
+			&constraint.Position,
+			&constraint.ReferencedTable,
+			&constraint.ReferencedColumn,
 		); err != nil {
 			return nil, err
 		}
 
-		constraints = append(constraints, constraint)
+		if constraints[constraint.ConstraintName] == nil {
+			constraints[constraint.ConstraintName] = []SpannerConstraint{}
+		}
+		constraints[constraint.ConstraintName] = append(constraints[constraint.ConstraintName], constraint)
 	}
 	if err = rows.Err(); err != nil {
 		return nil, err
 	}
-
 	return constraints, nil
 }
 
+
 func (h *spanner) dropAndRecreateConstraints(db *sql.DB, loadFn loadFunction) (err error) {
 	defer func() {
 		// Re-create constraints again after load
-		for _, constraint := range h.constraints {
+		for key := range h.constraints {
+			var lengthConstraints = len(h.constraints[key])
+			var orderedConstraints = make([]SpannerConstraint, lengthConstraints)
+
+			for _, constraint := range h.constraints[key] {
+				orderedConstraints[constraint.Position-1] = constraint
```

Comment:
please verify the position < len, cause it comes from db, so we don't control it